documentation-site/pages/components/avatar.js

```diff
@@ -0,0 +1,96 @@
+import Code from './code';
+// import {Avatar} from 'baseui/avatar';
+// import Avatar from '../../static/examples/avatar/sizes';
+
+// const code = `class ShoppingList extends React.Component {
+//   render() {
+//     return (
+//       <div className="shopping-list" style={{
+//         margin: '40px',
+//         border: '5px solid pink'
+//       }}>
+//         <h1>Shopping List for Mark</h1>
+//         <ul>
+//           <li>Instagram</li>
+//           <li>WhatsApp</li>
+//           <li>Oculus</li>
+//         </ul>
+//       </div>
+//     );
+//   }
+// }`;
+
+const code = `export default class Avatar extends React.Component<PropsT, StateT> {
+  static defaultProps: $Shape<PropsT> = {
+    overrides: {},
+    size: 'scale1000',
+  };
+
+  constructor(props: PropsT) {
+    super(props);
+    this.state = {noImageAvailable: !this.props.src};
+  }
+
+  handleError = () => {
+    this.setState({noImageAvailable: true});
+  };
+
+  render() {
+    const {noImageAvailable} = this.state;
+    const {name, overrides = {}, size, src} = this.props;
+    const [Avatar, avatarProps] = getOverrides(overrides.Avatar, StyledAvatar);
+    const [Initials, initialsProps] = getOverrides(
+      overrides.Initials,
+      StyledInitials,
+    );
+    const [Root, rootProps] = getOverrides(overrides.Root, StyledRoot);
+
+    return (
+      <Root
+        aria-label={noImageAvailable ? name : null}
+        role={noImageAvailable ? 'img' : null}
+        $didImageFailToLoad={noImageAvailable}
+        $size={size}
+        data-baseweb="avatar"
+        {...rootProps}
+      >
+        {noImageAvailable ? (
+          <Initials {...initialsProps}>{getInitials(name)}</Initials>
+        ) : (
+          <Avatar
+            alt={name}
+            onError={this.handleError}
+            src={src}
+            $size={size}
+            {...avatarProps}
+          />
+        )}
+      </Root>
+    );
+  }
+}`;
+
+// const code = `
```

Comment:
1) I tried to pass a simple example as `code`. Not surprisingly, it does not compile because `Avatar is not defined`.

Reply:
Right, you just need to pass it through the `scope` prop. Also, not sure if it can handle flow?